assembly/src/parsers/mod.rs

```diff
@@ -43,31 +43,27 @@ pub struct ProgramAst {
 
 impl ProgramAst {
     /// Returns byte representation of the `ProgramAst`.
-    pub fn to_bytes(&self) -> Result<Vec<u8>, SerializationError> {
-        let mut byte_writer = ByteWriter::default();
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut target = Vec::<u8>::default();
 
-        // local procedures
-        byte_writer.write_u16(self.local_procs.len() as u16);
+        target.write_u16(self.local_procs.len() as u16);
+        self.local_procs.write_into(&mut target);
 
-        self.local_procs.iter().try_for_each(|proc| proc.write_into(&mut byte_writer))?;
+        target.write_u64(self.body.len() as u64);
+        self.body.write_into(&mut target);
```

Comment:
I think `u64` both for body length is too much - let's use `u16` and add a TODO to enforce this restriction via the constructor.